src/rust/engine/BUILD

```diff
@@ -15,18 +15,20 @@ files(
     ],
 )
 
-shell_sources(name="shell")
+_RELTYPE_FLAG = "" if env("MODE") == "debug" else "--release"
 
 shell_command(
     name="engine-and-client",
-    command=f"./cargo_build_shim.sh --features=extension-module -p engine -p client",
+    command=(
+        'PATH="${PATH}:${CHROOT}/3rdparty/tools/python3/python/bin:${CHROOT}/3rdparty/tools/protoc/protoc/bin"'
+        + f" cargo build {_RELTYPE_FLAG} --features=extension-module -p engine -p client"
+    ),
     execution_dependencies=[
         ":rust_sources",
-        "./cargo_build_shim.sh:shell",
         "3rdparty/tools/protoc:protoc",
         "3rdparty/tools/python3:python3",
     ],
-    extra_env_vars=["CHROOT={chroot}", "MODE"],
+    extra_env_vars=["CHROOT={chroot}"],
     tools=["bash", "cc", "ld", "as", "ar", "cargo", "python3.9"],
```

Comment:
I didn't realize this still had `python3.9` in the `tools`. Whoopsie. Will do a follow-up PR